cmd/qat_plugin/dpdkdrv/dpdkdrv.go

```diff
@@ -62,11 +67,64 @@ var qatDeviceDriver = map[string]string{
 	"6f55": "d15xxvf",
 }
 
+func swapBDF(devstrings []string) []string {
+	result := make([]string, len(devstrings))
+
+	for n, dev := range devstrings {
+		tmp := strings.Split(dev, ":")
+		result[n] = fmt.Sprintf("%v:%v:%v", tmp[2], tmp[1], tmp[0])
+	}
+
+	return result
+}
+
+type preferredAllocationPolicyFunc func(*pluginapi.ContainerPreferredAllocationRequest) []string
+
+// nonePolicy is used for allocating QAT devices randomly.
+func nonePolicy(req *pluginapi.ContainerPreferredAllocationRequest) []string {
+	klog.V(2).Info("Select nonePolicy for QAT device allocation")
+
+	deviceIds := req.AvailableDeviceIDs[0:req.AllocationSize]
+
+	klog.V(2).Infof("Allocate deviceIds: %q", deviceIds)
+
+	return deviceIds
+}
+
+// balancedPolicy is used for allocating QAT devices in balance.
+func balancedPolicy(req *pluginapi.ContainerPreferredAllocationRequest) []string {
+	klog.V(2).Info("Select balancedPolicy for QAT device allocation")
+
+	// make it "FDB" and string sort
+	deviceIds := swapBDF(req.AvailableDeviceIDs)
+	sort.Strings(deviceIds)
+	deviceIds = swapBDF(deviceIds)
+
+	// return "BDF"
+	return deviceIds[:req.AllocationSize]
+}
+
+// packedPolicy is used for allocating QAT devices one by one.
```

Comment:
let's update the comments for all functions with `PF` added:
```suggestion
// packedPolicy is used for allocating QAT PF devices one by one.
```